### PR TITLE
MMT-1018 UMM form inline schema validation

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: app/assets/javascripts/validation.coffee [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+      json = $('.umm-form').find('input, textarea, select').filter ->
+        return this.value
+      json = JSON.parse(json.serializeJSON())
+      if json.ServiceDraft?
+        json = json.ServiceDraft.Draft
+      else if json.VariableDraft?
+        json = json.VariableDraft.Draft
+      else
+        json = {}
+      fixRelatedURL(json)
```
Comment (macrouch) [inline app/assets/javascripts/validation.coffee]: move into `if json.ServiceDraft?`

File: lib/json_schema_form/umm_form.rb [hunk 2]
```diff
@@ -74,7 +74,8 @@ def render_markup
 
       if parsed_json['label']
         label_text = parsed_json['label']
-        concat label_tag('', label_text, class: ('eui-required-o' if schema.required_field?(full_key)))
+        # concat label_tag('', label_text, class: ('eui-required-o' if schema.required_field?(full_key)))
```
Comment (macrouch) [inline lib/json_schema_form/umm_form.rb]: remove

File: lib/json_schema_form/umm_form.rb [hunk 3]
```diff
@@ -448,18 +449,18 @@ def validation_properties(element)
...
         'msg-required': "#{schema.fetch_key_leaf(element['key']).titleize} is required"
       }
     end
 
     if element['type'] == 'number'
-      validation_properties['number'] = true
+      validation_properties[:number] = true
 
-      validation_properties['data'] = validation_properties.fetch('data', {}).merge(
+      validation_properties[:data] = validation_properties.fetch(:data, {}).merge(
```
Comment (macrouch) [inline lib/json_schema_form/umm_form.rb]: do we need these attributes?

File: lib/json_schema_form/umm_form.rb [hunk 4]
```diff
@@ -532,7 +542,10 @@ def render_markup
       unless form_fragment['noLabel']
         label_text = form_element.title
         label_text = parsed_json['label'] if parsed_json['label']
-        concat label_tag(keyify_property_name, label_text, class: ('eui-required-o' if schema.required_field?(full_key)))
+        # concat label_tag(keyify_property_name, label_text, class: ('eui-required-o' if schema.required_field?(full_key)))
```
Comment (macrouch) [inline lib/json_schema_form/umm_form.rb]: remove

File: app/assets/javascripts/validation.coffee [hunk 5]
```diff
@@ -99,17 +147,17 @@ $(document).ready ->
...
+    if classes.indexOf('mmt-integer') != -1 or $(element).attr('integer') == 'true'
       type = 'integer'
-    if classes.indexOf('mmt-boolean') != -1
+    if classes.indexOf('mmt-boolean') != -1 or $(element).attr('boolean') == 'true'
       type = 'boolean'
-    if classes.indexOf('mmt-date-time') != -1
+    if classes.indexOf('mmt-date-time') != -1 or $(element).attr('date-time') == 'true'
       type = 'date-time'
-    if classes.indexOf('mmt-uri') != -1
+    if classes.indexOf('mmt-uri') != -1 or $(element).attr('uuid') == 'true'
```
Comment (charleshuang80) [inline app/assets/javascripts/validation.coffee]: I think this is the wrong `attr`.
also, I know we've mentioned this but I don't think we've come to a definitive answer, do we want to use `prop` instead? https://stackoverflow.com/questions/5874652/prop-vs-attr

File: lib/json_schema_form/umm_form.rb [hunk 6]
```diff
@@ -444,25 +444,12 @@ def idify_property_name(ignore_keys: %w(items properties index_id))
   def validation_properties(element)
     # jQuery Validate can use html elements for validation so we'll set those elements
     # here instead of having to define these attributes on a one off basis in javascript
-    validation_properties = element.select { |key| %w(minLength maxLength pattern).include?(key) }
+    validation_properties = element.select { |key| %w(maxLength).include?(key) }
 
     # JSON Schema provides the required fields in a separate array so we have to look this up
-    if schema.required_field?(element['key'])
-      validation_properties['required'] = true
+    # validation_properties[:required] = true if schema.required_field?(element['key'])
```
Comment (charleshuang80) [inline lib/json_schema_form/umm_form.rb]: is this supposed to be commented out?

**Comment by charleshuang80** [comment]
approving, pending green build

**Pull request merged**